improver/developer_tools/metadata_interpreter.py

```diff
@@ -0,0 +1,563 @@
+                msg = (
+                    f"no coord with var_name='threshold' found in all coords: {coords}"
+                )
+                self.errors.append(msg)
+            else:
+                msg += f"threshold coord has incorrect name '{threshold_name}'"
+                self.errors.append(msg)
+                self.check_threshold_coordinate_properties(
+                    cube.name(), cube.coord(threshold_name)
+                )
+        else:
+            threshold_coord = cube.coord(expected_threshold_name)
+            self.check_threshold_coordinate_properties(cube.name(), threshold_coord)
+
+    def check_threshold_coordinate_properties(self, cube_name, threshold_coord):
+        """Checks threshold coordinate properties are correct and consistent with
+        cube name"""
+        threshold_var_name = threshold_coord.var_name
+        if threshold_var_name != "threshold":
+            self.errors.append(
+                f"Threshold coord {threshold_coord.name()} does not have "
+                "var_name='threshold'"
+            )
+
+        self.relative_to_threshold = threshold_coord.attributes[
+            "spp__relative_to_threshold"
+        ]
+
+        if self.relative_to_threshold in ("greater_than", "greater_than_or_equal_to"):
+            threshold_attribute = "above"
+        elif self.relative_to_threshold in ("less_than", "less_than_or_equal_to"):
+            threshold_attribute = "below"
+        elif self.relative_to_threshold == "between_thresholds":
+            threshold_attribute = "between"
+            self.warnings.append("Between thresholds data are not fully supported")
+        else:
+            threshold_attribute = None
+            self.errors.append(
+                f"spp__relative_to_threshold attribute '{self.relative_to_threshold}' "
+                "is not in permitted value set"
+            )
+
+        if threshold_attribute and threshold_attribute not in cube_name:
+            self.errors.append(
+                f"Cube name '{cube_name}' is not consistent with "
+                f"spp__relative_to_threshold attribute '{self.relative_to_threshold}'"
+            )
+
+    def check_cell_methods(self, cube):
+        """Checks cell methods are permitted and correct"""
+        cell_methods = cube.cell_methods
+        for cm in cell_methods:
+            if cm.method in COMP_CM_METHODS:
+                self.methods += f" {cm.method} over {cm.coord_names[0]}"
+                if self.field_type == self.PROB:
+                    if not cm.comments or cm.comments[0] != f"of {self.diagnostic}":
+                        self.errors.append(
+                            f"Cell method {cm} on probability data should have comment "
+                            f"'of {self.diagnostic}'"
+                        )
+                # check point and bounds on method coordinate
+                if "time" in cm.coord_names:
+                    if cube.coord("time").bounds is None:
```

Comment:
This section feels out of place. We're checking cell methods, but here the code goes and checks the bounds of the time coordinate. I think this method should just check that the time coord HAS bounds and that a separate method should check that any time coord with bounds has points that are set at the upper bound.

Reply:
Well, OK, I can see your point.  I'm not sure what that does to the logic and code verbosity / complexity, but I'll try to reorganise this a little.

Reply:
OK, this was simpler than I feared.  Done now.